src/Orleans.Runtime/Hosting/CoreHostingExtensions.cs

```diff
@@ -95,11 +110,62 @@ public static ISiloHostBuilder ConfigureSiloName(this ISiloHostBuilder builder,
         }
 
         /// <summary>
-        /// Configure silo to use Development membership
+        /// Configures a localhost silo for development and testing.
         /// </summary>
-        public static ISiloHostBuilder UseDevelopmentClustering(this ISiloHostBuilder builder, Action<DevelopmentMembershipOptions> configureOptions)
+        /// <param name="builder">The silo builder.</param>
+        /// <param name="siloPort">The silo port.</param>
+        /// <param name="gatewayPort">The gateway port.</param>
+        /// <param name="primarySiloEndpoint">
+        /// The endpoint of the primary silo, or <see langword="null"/> to use this silo as the primary.
+        /// </param>
+        /// <returns>The silo builder.</returns>
+        public static ISiloHostBuilder UseLocalhostClustering(
+            this ISiloHostBuilder builder,
+            int siloPort = EndpointOptions.DEFAULT_SILO_PORT,
+            int gatewayPort = EndpointOptions.DEFAULT_GATEWAY_PORT,
+            IPEndPoint primarySiloEndpoint = null,
+            string clusterId = ClusterOptions.DevelopmentClusterId)
         {
-            return builder.ConfigureServices(
+            builder.Configure<EndpointOptions>(options =>
+            {
+                options.AdvertisedIPAddress = IPAddress.Loopback;
+                options.SiloPort = siloPort;
+                options.GatewayPort = gatewayPort;
+            });
+
+            builder.UseDevelopmentClustering(primarySiloEndpoint ?? new IPEndPoint(IPAddress.Loopback, siloPort));
+
+            if (!string.IsNullOrWhiteSpace(clusterId))
+            {
+                builder.Configure(options => options.ClusterId = clusterId);
```

Comment:
hmm if clusterId is null, which means user explicitly set it to null, I don't think we should just ignore their choice without telling them, leaving them the impression that clusterId is not needed. We should do a ConfigurationValidaiton instead

Reply:
Perhaps we set a default value for ClusterId if it's not set by the user, similar to how we handle SiloName